lib/Serialization/DeserializeSIL.cpp

```diff
@@ -460,7 +494,15 @@ SILDeserializer::readSILFunctionChecked(DeclID FID, SILFunction *existingFn,
 
   SmallVector<uint64_t, 64> scratch;
   StringRef blobData;
-  unsigned kind = SILCursor.readRecord(entry.ID, scratch, &blobData);
+  llvm::Expected<unsigned> maybeKind =
+      SILCursor.readRecord(entry.ID, scratch, &blobData);
+  if (!maybeKind) {
+    LLVM_DEBUG(llvm::dbgs() << "Cursor advance error in readSILFunction: "
+                            << toString(maybeKind.takeError()) << "\n");
+    MF->error();
```

Comment:
Same here and below: `MF->error()` is going to abort anyway, so you might as well use `MF->fatal(maybeKind.takeError())`.

Reply:
These are all going to be a problem in non-asserts builds, right? Because the `LLVM_DEBUG` will be compiled out? Or is that okay because it happens to line up with the checking of cleanup being compiled out?

Reply:
It will be a problem in debug, yes. I'm trying to match current behavior and / or surrounding code. I'll update as you suggest.

Reply:
Sorry, I mean because the error won't be taken in release builds if you use `LLVM_DEBUG`.

Reply:
Ah yeah, that too.